smolyay/basis.py

```diff
@@ -72,6 +84,32 @@ def in_domain(self, x):
             numpy.greater_equal(x, self.domain[0]), numpy.less_equal(x, self.domain[1])
         )
 
+    def _scale_to_domain(self, points, old_domain):
+        """Scale points from a domain to BasisFunction domain
+
+        Parameters
+        ----------
+        points: numeric or array-like
+            points to be shifted to new domain.
+
+        old_domain: ndarray of shape (2,)
+            upper and lower bounds of points.
+
+        Returns
+        -------
+        numeric or array-like
+            points shifted to BasisFunction domain"""
+        points = numpy.array(points)
+        new_points = numpy.asarray(
```

Comment:
Due to the cast of `points` above, you should not need `asarray` here
```suggestion
        new_points = (
```

Reply:
Weirdly, it is needed. If I input a single number into this function, then numpy.clip will raise "TypeError: return arrays must be of ArrayType" and the cast with asarray fixes it.

Getting rid of `points = numpy.array(points)` doesn't cause any issues, fortunately.

Reply:
I think you need to have the cast of `points` (what if someone gives a list?) You might need to add a minimum dimension to ensure it casts up a scalar though.